cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -116,16 +116,17 @@ def _invalid_endpoint(self, endpoint_str):
         return False
 
     def endpoint_dial(self, endpoint_str):
-        try:
-            ep = "http://" + endpoint_str
-            r = requests.head(ep)
-            rc = r.status_code
-            if rc != 200:
-                raise ExecutionFailureException(
-                    "wrong return code {} on rgw endpoint http header request".format(rc))
-        except requests.ConnectionError:
-            raise ExecutionFailureException(
-                "failed to connect to rgw endpoint {}".format(ep))
+        protocols = ["http", "https"]
+        for prefix in protocols:
+            try:
+                ep = "{}://{}".format(prefix, endpoint_str)
+                r = requests.head(ep)
```

Comment:
What's the timeout all ready for this? Can we shorten it to a few seconds? Like <5s?

Reply:
I have added 3 second as per the following [doc](https://github.com/psf/requests/blob/master/requests/api.py),
```
:param timeout: (optional) How many seconds to wait for the server to send data
        before giving up, as a float, or a :ref:`(connect timeout, read timeout) <timeouts>` tuple.
        :type timeout: float or tuple
```

Reply:
Did you test your change?

Reply:
Yes I have tested my change individually (by providing URLs like golang , youtube websites and other non-existing) on a separate python file. It works within the timeout.

```python
import sys
import requests

class ExecutionFailureException(Exception):
    pass

def endpoint_check(endpoint_str, timeout=5):
    protocols = ["http", "https"]
    for prefix in protocols:
        try:
            ep = "{}://{}".format(prefix, endpoint_str)
            print("EP: {}".format(ep))
            r = requests.head(ep, timeout=timeout)
            if r.status_code == 200:
                return
            print("EP Status: {}".format(r.status_code))
        except:
            continue
    raise ExecutionFailureException(
        "unable to connect to endpoint: {}".format(endpoint_str))


def main():
    endpoint = "vlang.io"
    if len(sys.argv) > 1:
        endpoint = sys.argv[1]
    endpoint_check(endpoint, timeout=3)

if __name__ == "__main__":
    main()
```